### PR TITLE
Fix for input value lists to be represented as IEnumerable

### LINKED ISSUE #8: Input values should not be passed in as an `IQueryable`.
For example [Repository.Issues](https://github.com/grokys/Octokit.GraphQL/blob/master/Octokit.GraphQL/Repository.cs#L132) has `IQueryable<IssueState?> states` - this should be e.g. `IEnumerable<IssueState>`.

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Fixes #8 

Depends on: 
- [x] #19
- [x] #26

### CODE REVIEW

File: Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs [hunk 1]
```diff
@@ -36,6 +36,18 @@ public static string GetCSharpType(TypeModel type)
...
 
+        public static string GetCSharpArgType(TypeModel type)
+        {
+            //TODO: Do this, but not dirty
+            switch (type.Kind)
+            {
+                case TypeKind.List:
+                    return $"IEnumerable<{GetCSharpType(type.OfType, false)}>";
+                default:
+                    return GetCSharpType(type);
```
Comment (StanleyGoldman) [inline Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs]: `GetCSharpType()` returns `IQueryable` and send `true` to the underlying type call, which puts a `?` after `Enum`s

File: Octokit.GraphQL.Core.Generation/EntityGenerator.cs [hunk 2]
```diff
@@ -237,7 +237,7 @@ private static void GenerateArguments(FieldModel field, out string arguments, ou
                 }
 
                 var argName = TypeUtilities.GetArgName(arg);
-                argBuilder.Append(TypeUtilities.GetCSharpType(arg.Type));
+                argBuilder.Append(TypeUtilities.GetCSharpArgType(arg.Type));
```
Comment (StanleyGoldman) [inline Octokit.GraphQL.Core.Generation/EntityGenerator.cs]: This is the call to generate the types for function parameters

File: Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs (hunk 1 above)
Comment (grokys) [inline Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs]: I don't know, this looks fine to me? `GetCSharpType` probably shouldn't be passing `true` to the underlying type call though? But that's a separate issue.

File: Octokit.GraphQL.Core.Generation.UnitTests/InterfaceGenerationTests.cs [hunk 3]
```diff
@@ -299,6 +301,74 @@ public void Generates_Method_For_List_Field_With_Args()
...
+                            }
+                        }
+                    },
+                }
+            };
+
+            var result = CodeGenerator.Generate(model, "Test", null);
+
+            Assert.Equal(expected, result);
+        }
```
Comment (grokys) [inline Octokit.GraphQL.Core.Generation.UnitTests/InterfaceGenerationTests.cs]: Might be worth adding a test here for non-null list args?

File: Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs (hunk 1 above)
Comment (StanleyGoldman) [inline Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs]: Cool, yea, this is a target for a refactor at some point.

**Comment by StanleyGoldman** [comment]
Added those tests

File: Octokit.GraphQL.Core.Generation.UnitTests/InterfaceGenerationTests.cs [hunk 4]
```diff
@@ -335,6 +369,40 @@ public void Generates_Method_For_List_Field_With_Object_List_Arg()
         }
 
         [Fact]
+        public void Generates_Method_For_List_Field_With_NonNull_Object_List_Arg()
+        {
+            var expected = FormatMemberTemplate(
+                "IQueryable<Other> Foo(IEnumerable<Another> bar = null);",
```
Comment (grokys) [inline Octokit.GraphQL.Core.Generation.UnitTests/InterfaceGenerationTests.cs]: If the arg isn't nullable, then it probably shouldn't have a `null` default value!

Comment (StanleyGoldman) [inline Octokit.GraphQL.Core.Generation.UnitTests/InterfaceGenerationTests.cs]: Ah, i knew I was missing something obvious.

File: Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs [hunk 5]
```diff
@@ -101,12 +106,12 @@ public static TypeModel ReduceType(TypeModel type)
             return type;
         }
 
-        private static string GetCSharpType(TypeModel type, bool nullable)
+        private static string GetCSharpType(TypeModel type, bool nullableType, bool returnType)
```
Comment (StanleyGoldman) [inline Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs]: This makes a lot more sense now

File: Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs [hunk 6]
```diff
@@ -31,9 +31,14 @@ public static string GetCSharpLiteral(string value, TypeModel type)
             throw new NotImplementedException();
         }
 
-        public static string GetCSharpType(TypeModel type)
+        public static string GetCSharpReturnType(TypeModel type)
```
Comment (StanleyGoldman) [inline Octokit.GraphQL.Core.Generation/Utilities/TypeUtilities.cs]: And this method has a clearer purpose now.
